apps/roam/src/components/settings/AdminPanel.tsx

```diff
@@ -211,6 +208,147 @@ const AdminPanel = (): React.ReactElement => {
     return <p className="text-red-700">{error}</p>;
   }
 
+  return (
+    <>
+      <p>
+        Context:{" "}
+        <code>{JSON.stringify({ ...context, spacePassword: "****" })}</code>
+      </p>
+      {schemas.length > 0 ? (
+        <>
+          <Label>
+            Display:
+            <div className="mx-2 inline-block">
+              <Select
+                items={schemas}
+                onItemSelect={(choice) => {
+                  setShowingSchema(choice);
+                }}
+                itemRenderer={(node, { handleClick, modifiers }) => (
+                  <MenuItem
+                    active={modifiers.active}
+                    key={node.sourceLocalId}
+                    label={node.name}
+                    onClick={handleClick}
+                    text={node.name}
+                  />
+                )}
+              >
+                <Button text={showingSchema.name} />
+              </Select>
+            </div>
+          </Label>
+          <div>{loadingNodes ? <Spinner /> : <NodeTable nodes={nodes} />}</div>
+        </>
+      ) : (
+        <p>No node schemas found</p>
+      )}
+    </>
+  );
+};
+
+const MigrationTab = (): React.ReactElement => {
+  let initial = true;
+  const enabled = getSetting("use-reified-relations");
+  const [useMigrationResults, setMigrationResults] = useState<string>("");
+  const [useOngoing, setOngoing] = useState<boolean>(false);
+  const [useDryRun, setDryRun] = useState<boolean>(false);
+  const doMigrateRelations = async () => {
+    setOngoing(true);
+    try {
+      const before = await countReifiedRelations();
+      const numProcessed = await migrateRelations(useDryRun);
+      const after = await countReifiedRelations();
+      if (after - before < numProcessed)
+        setMigrationResults(
+          `${after - before} new relations created out of ${numProcessed} distinct relations processed`,
+        );
+      else setMigrationResults(`${numProcessed} new relations created`);
+    } catch (e) {
+      console.error("Relation migration failed", e);
+      setMigrationResults(
+        `Migration failed: ${(e as Error).message ?? "see console for details"}`,
+      );
+    } finally {
+      setOngoing(false);
+    }
+  };
+  useEffect(() => {
+    void (async () => {
+      if (initial) {
+        const numRelations = await countReifiedRelations();
+        setMigrationResults(
+          numRelations > 0
+            ? `${numRelations} already migrated`
+            : "No migrated relations",
+        );
+        initial = false;
+      }
```

Comment:
"Assignments to the 'initial' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect.eslintreact-hooks/exhaustive-deps"

Reply:
This is the comment raised by the IDE.  If you plan on leaving that code in there, add the appropriate comment to ignore the warning.